File: 07Expression/expression.rs
#[allow(unused)]
fn main() {
    let x = 5u32;
	//这个类似c语言的逗号，最后一个是结果。顺序是从左到右。
    let y = {
        let x_squared = x * x;
        let x_cube = x_squared * x;

        // 将此表达式赋给 `y`
        x_cube + x_squared + x
    };
	//默认是unit类型，表示为空的tuple。这是Rust特殊的。
    let z = {
        // 分号结束了这个表达式，于是将 `()` 赋给 `z`
        2 * x;
    };

    println!("x is {:?}", x);
    println!("y is {:?}", y);
    println!("z is {:?}", z);
}

File: 07Expression/expression.cs
using System;
public class Program
{
    public static void Main(string[] args)
    {
		Console.WriteLine("这是c#语言");
        Console.WriteLine("世界你好!");
        var x = 5u;
        // 使用匿名方法同时声明并调用 比rust里面要麻烦多了
        var y = new Func<uint>(() =>
        {
            var x_squared = x * x;
            var x_cube = x_squared * x;

            // 将此表达式赋给 `y`
            return x_cube + x_squared + x;
        })();
        var z = 2*x;
        Console.WriteLine("x is {0}", x);
		Console.WriteLine("y is {0}", y);
		Console.WriteLine("z is {0}", z);
    }
}